clang/include/clang/Driver/Options.td

```diff
@@ -1967,6 +1967,10 @@ def fclang_abi_compat_EQ : Joined<["-"], "fclang-abi-compat=">, Group<f_clang_Gr
   Visibility<[ClangOption, CC1Option]>,
   MetaVarName<"<version>">, Values<"<major>.<minor>,latest">,
   HelpText<"Attempt to match the ABI of Clang <version>">;
+def fno_demangling_failures: Flag<["-"], "fno-demangling-failures">, Group<f_clang_Group>,
```

Comment:
I don't recall how it works, but I think we want an on & off flag type, not just the off.

Reply:
Yeah this flag turns on an assertion that validates that all the manglings produced by clang can be demangled. In this sense it on & off (off by default). Let me know if I misunderstand something.

Reply:
In that we need a 'turn it back on' flag as well, so we need `-fdemangling-failures`.

Reply:
This has turned into `-fdiagnostics-demangler-failures` now.